logdetective/server/server.py

```diff
@@ -558,11 +566,13 @@ async def retrieve_and_preprocess_koji_logs(job):
 
     LOG.debug("Failed architecture: %s", failed_arch)
 
-    log_path = failed_arches[failed_arch]
-    LOG.debug("Returning contents of %s", log_path)
+    log_path = failed_arches[failed_arch].as_posix()
+
+    log_url = f"{SERVER_CONFIG.gitlab.api_url}/projects/{job.project_id}/jobs/{job.id}/artifacts/{log_path}"
```

Comment:
nit: I'm surprised the python gitlab bindings don't provide a function to construct this URL

Reply:
They allow us to directly retrieve the contents of the file, so providing us the URL itself is probably not something the python-gitlab developers thought would be particularly useful.

Also, that's not a direct URL; I think that returns a 302 redirect to a load-balanced, geolocated file server.